reconcile/unleash_feature_toggles/integration.py

```diff
@@ -0,0 +1,310 @@
+            desired=[d for d in desired_state if d.delete is not True],
+            current_key=lambda c: c.name,
+            desired_key=lambda d: d.name,
+            equal=feature_toggle_equal,
+        )
+        for add in diff.add.values():
+            logging.info(
+                f"[{instance.name}/{project_id}] Adding feature toggle {add.name}"
+            )
+            try:
+                assert add.unleash.q_type  # make mypy happy
+                feature_type = FeatureToggleType[add.unleash.q_type]
+            except KeyError:
+                raise ValueError(
+                    f"[{instance.name}/{project_id}/{add.name}] Invalid feature toggle type '{add.unleash.q_type}', Possible values are: {', '.join(FeatureToggleType.__members__)}"
+                )
+            if not dry_run:
+                client.create_feature_toggle(
+                    project_id=project_id,
+                    name=add.name,
+                    description=add.description,
+                    type=feature_type,
+                    impression_data=bool(add.unleash.impression_data),
+                )
+
+        for change in diff.change.values():
+            logging.info(
+                f"[{instance.name}/{project_id}] Changing feature toggle {change.desired.name}"
+            )
+            try:
+                assert change.desired.unleash.q_type  # make mypy happy
+                feature_type = FeatureToggleType[change.desired.unleash.q_type]
+            except KeyError:
+                raise ValueError(
+                    f"[{instance.name}/{project_id}/{change.current.name}] Invalid feature toggle type '{change.desired.unleash.q_type}', Possible values are: {', '.join(FeatureToggleType.__members__)}"
+                )
+            if not dry_run:
+                client.update_feature_toggle(
+                    project_id=project_id,
+                    name=change.current.name,
+                    description=change.desired.description,
+                    type=feature_type,
+                    impression_data=bool(change.desired.unleash.impression_data),
+                )
+
+        for delete in diff.delete.values():
+            desired_toggle = next(
+                (d for d in desired_state if d.name == delete.name and d.delete), None
+            )
+            if desired_toggle:
+                logging.info(
+                    f"[{instance.name}/{project_id}] Deleting feature toggle {delete.name}"
+                )
+                if not dry_run:
+                    client.delete_feature_toggle(
+                        project_id=project_id,
+                        name=delete.name,
+                    )
+            elif not instance.allow_unmanaged_feature_toggles:
+                raise UnleashFeatureToggleDeleteError(
+                    f"[{instance.name}/{project_id}] Found unmanaged feature toggles '{[d.name for d in diff.delete.values()]}'"
+                )
```

Comment:
this will show all `diff.delete.values()`, some of them may already included in the previous `if desired_toggle` branch.

Reply:
We'll hit that in a PR check, and only if someone deletes a managed toggle and there are unmanaged toggles and unmanaged toggles aren't allowed. This is an edge case and can be ignored.